chapter_15_simple_form.asciidoc

```diff
@@ -1183,6 +1198,16 @@ so you always need the server side as well
 if you really care about data integrity;
 using a form is a nice way of encapsulating that logic.
 
+// DAVID: I wonder if this is slightly missing the point.
+// Our functional tests here are very high level - using a headless browser.
+// But many developers use the Django test client
+// (https://docs.djangoproject.com/en/5.1/topics/testing/tools/#the-test-client)
+// for this kind of thing, which doesn't have browser validation built in.
+// So we could use that if we really wanted to test the server side validation.
```

Comment:
Update: I just clocked that we are doing exactly this in `test_views.py`. Might be worth just calling it out a bit more?